src/averageAges.js

```diff
@@ -55,7 +71,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.reduce((arr, person) => {
+    arr.push(
+      {
+        ...person,
+        mother: people.find(mother => mother.name === person.mother),
+      },
+    );
+
+    return arr;
+  }, []);
+
+  const agesList = child
+    .filter(onlyWithSon
+      ? person => person.sex === 'm' && person.mother !== undefined
+      : person => person.mother !== undefined)
+    .map(person => person.born - person.mother.born);
+
+  return agesList.reduce(
```

Comment:
Here you don't use your `getAverageAge` function. You can change it to make it suitable for all cases